rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/310_date_agg_per_day_of_week.yml

```diff
@@ -0,0 +1,44 @@
+
+setup:
+  - skip:
+      version: " - 7.5.99" #todo this test failing on jdk8. has to be removed from 7.x
+      reason:  "Start of the week Monday was backported to 7.5"
+
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            properties:
+              date:
+                type: date
+  - do:
+      index:
+        index: test
+        id:    1
+        body:  { "date": "2009-11-15T14:12:12" }
+  - do:
+      indices.refresh:
+        index: [test]
+---
+# The inserted document has a field date=2009-11-15T14:12:12 which is Sunday.
+# When aggregating per day of the week this should be considered as last day of the week (7)
+# and this value should be used in 'key_as_string'
+"Date aggregartion per day of week":
```

Comment:
```suggestion
"Date aggregation per day of week":
```

Reply:
yes - as per todo this test is expected to fail in 1.8 https://github.com/elastic/elasticsearch/pull/50916/files#diff-a1dab819a2bedbcb781ecd85580cfbb5R4
I am actually not sure how to work around this, the code for using IsoCalendarDataProvider is in 7.x but because we support JDK1.8 I won't be able to always pass the build. 
Any suggestions how to fix this?
I am thinking of just removing the test for now..

Reply:
@pugnascotia @rjernst what is your view on this? Maybe we should extend the testing framework to allow checking runtime jvm and skip basing on this?

Reply:
My feeling is that these changes are fine, but what I'm missing is some Javadoc to explain why they're necessary. Unless I have all the context around dates and JDK versions, I can't understand what the checks are for. The new feature name doesn't help here either. Can we add some more explanation into the code, to help other developers understand what is going on and why?

Reply:
@pugnascotia good idea. will add some more javadocs as they are clearly missing